src/Build/Evaluation/Evaluator.cs

```diff
@@ -811,13 +812,28 @@ private void Evaluate()
             if (this._evaluationLoggingContext.LoggingService.IncludeEvaluationPropertiesAndItems)
             {
                 globalProperties = _data.GlobalPropertiesDictionary;
-                properties = _data.Properties;
+                properties = Traits.Instance.LogAllEnvironmentVariables ? _data.Properties : FilterOutEnvironmentDerivedProperties(_data.Properties);
                 items = _data.Items;
             }
 
             _evaluationLoggingContext.LogProjectEvaluationFinished(globalProperties, properties, items, _evaluationProfiler.ProfiledResult);
         }
 
+        private IEnumerable FilterOutEnvironmentDerivedProperties(PropertyDictionary<P> dictionary)
+        {
+            List<P> list = new();
+            foreach (P p in dictionary)
+            {
+                if (!((p is ProjectProperty pp && pp.IsEnvironmentProperty) ||
```

Comment:
this condition is confusing to read. Is there a way to simplify this and remove unnecessary parentheses? Potentially also invert it and call `continue;` if false?